### DISCUSSION TITLE
VRP: load constraint exceeded with float digits

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2852] [Routing (and legacy CP) questions] [answered]
cross-posted here: https://groups.google.com/g/or-tools-discuss/c/oqMgzvA0BAU

load (cubic meters) is one of the containts defining the vehicle capacity in my model. 

this is how I set this load containt:
```py
def add_cm3_constraints(routing, data, demand_callback_index):
	"""Adds capacity constraint"""
	cm3 = "cm3"
	routing.AddDimensionWithVehicleCapacity(
		demand_callback_index,
		0, # null capacity slack
		cm3_vector(), # vector vehicle_capacity
		True, # start cumul to zero
		cm3)
```
If I set the vehicle with max load capacity being as "30", the consolidated loads in my vehicle should not exceed "30".
However, I often get the actual load of the vehicle exceeding a little bit, like: 31.45 or 30. 78. The individual loads are float number but I do not think this will affect the max load limit set above. what is the root cause of this error?

**UPDATE:**
I found "self.assignment.Value(cm3_var)" is actually equal to "math.floor(cm3_var)", hence all the loads (cm3) are floored in the model. why does this happen? does ortools only accept the integer value?

**Comment by Mizux** [comment, marked as answer]
yes routing solver in C++ use int64_t all python,java, .Net module are "just" wrapper on top of the C++ library with the same limitation.
i.e. our python package is a native python package...
